sky/cli.py

```diff
@@ -3471,11 +3474,21 @@ def storage_delete(names: List[str], all: bool):  # pylint: disable=redefined-bu
     if sum([len(names) > 0, all]) != 1:
         raise click.UsageError('Either --all or a name must be specified.')
     if all:
-        click.echo('Deleting all storage objects.')
         storages = sky.storage_ls()
         names = [s['name'] for s in storages]
     else:
         names = _get_glob_storages(names)
+    if not yes and len(names) > 0:
+        storage_names = ', '.join(names)
+        storage_str = 'storages' if len(names) > 1 else 'storage'
+        click.confirm(
+            f'Deleting {len(names)} {storage_str}: '
+            f'{storage_names}. Proceed?',
+            default=True,
+            abort=True,
+            show_default=True)
+    elif all and len(names) == 0:
+        click.echo('No storage(s) to delete.')
```

Comment:
```suggestion
    if names:
        if not yes:
            storage_names = ', '.join(names)
            storage_str = 'storages' if len(names) > 1 else 'storage'
            click.confirm(
                f'Deleting {len(names)} {storage_str}: '
                f'{storage_names}. Proceed?',
                default=True,
                abort=True,
                show_default=True)
    elif all:
        click.echo('No storage(s) to delete.')
```
nit: refactored the conditions for readability and efficiency removing redundant `len(names)` check

Reply:
Great, thank you so much for the review!